### PR TITLE
KB Article: AddressableEntity

### PR DESCRIPTION
**Pull request by sczembor**

### CODE REVIEW

File: articles/XXX-addressable_entity.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# AddressableEntity in Casper 2.0
+
+Casper 2.0 introduces significant changes in the representation and management of accounts and smart contracts, through the introduction of the `AddressableEntity` type. This new structure replaces the separate `AccountHash` and `ContractHash` used in Casper 1.x, bringing a unified approach to interact with entities on the network.
```
[the remark this answers is not in the document]
Comment (sczembor) [reply articles/XXX-addressable_entity.md]: Done

File: articles/XXX-addressable_entity.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+Post-Condor, all accounts and smart contract addresses residing within the global state will be automatically migrated to the `AddressableEntity` structure. This means the network itself will recognize and handle these entities using the new format.
+
+**Smart Contract Responsibility:**
+
+While the global state transitions is automatic, the internal mechanisms of your smart contracts might not be. If your contracts rely on identifying the caller (e.g., to enforce permissions) or store addresses in their own internal storage, you'll need to take action.
+
+**Essential Steps for Contract Compatibility:**
+
+1. **Internal Storage Migration:** Update any data stored within your contract that uses the old `AccountHash` or `ContractHash` format. This typically involves writing custom migration functions to convert these keys to the new `Key::AddressableEntity` structure. The CEP-18 contract provides excellent examples (`migrate_user_balance_keys`, `migrate_user_allowance_keys`) that can serve as templates for your own migration logic.
```
[the remark this answers is not in the document]
Comment (sczembor) [reply articles/XXX-addressable_entity.md]: I reworked that section removing the migrations steps all together. Just added few considerations for the developers

File: articles/XXX-addressable_entity.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+2. **Caller Identification:** If your contract's logic depends on distinguishing between accounts and contracts calling it, you'll need to update your code to work with the new `Key::AddressableEntity` format.
+
+3. **External Contract Interactions:**  If your contract interacts with other contracts, verify that the parameters you pass when calling their entry points correspond to the new `Key::AddressableEntity` format.
+
+**Important Note:**
+
+* Failing to upgrade your contract could lead to unexpected behavior or errors, especially if your logic relies on the distinction between accounts and contracts.
```
[the remark this answers is not in the document]
Comment (sczembor) [reply articles/XXX-addressable_entity.md]: Done

File: articles/XXX-addressable_entity.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+**Smart Contract Responsibility:**
+
+While the global state transitions is automatic, the internal mechanisms of your smart contracts might not be. If your contracts rely on identifying the caller (e.g., to enforce permissions) or store addresses in their own internal storage, you'll need to take action.
+
+**Essential Steps for Contract Compatibility:**
+
+1. **Internal Storage Migration:** Update any data stored within your contract that uses the old `AccountHash` or `ContractHash` format. This typically involves writing custom migration functions to convert these keys to the new `Key::AddressableEntity` structure. The CEP-18 contract provides excellent examples (`migrate_user_balance_keys`, `migrate_user_allowance_keys`) that can serve as templates for your own migration logic.
+
+2. **Caller Identification:** If your contract's logic depends on distinguishing between accounts and contracts calling it, you'll need to update your code to work with the new `Key::AddressableEntity` format.
```
[the remark this answers is not in the document]
Comment (sczembor) [reply articles/XXX-addressable_entity.md]: See my response above

**Review by melpadden — APPROVED** [review]
Looks good. Any further changes can be in a separate CR.

**Pull request merged**